### PR TITLE
Issue #1 - Read from and write to the Firestore database

### LINKED ISSUE #1: 1. As a component, I want to be able to read from and write to the Firestore database so that users can persist information.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

AC:

* The following have been added as project dependencies: `firebase` & `react-firestore`
* Make a change in the Firestore database and it shows up in the app
* Make a change in the app and it shows up in the Firestore database

### PR DESCRIPTION
**Pull request by hajarNasr**
## Type of Changes

|      | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: New feature     |
|  ✓    | :link: Update dependencies |

## Acceptance Criteria

- [X] The following have been added as project dependencies: `firebase` & `react-firestore`
- [X] Make a change in the Firestore database and it shows up in the app
- [X] Make a change in the app and it shows up in the Firestore database

## Related Issue
Closes #1 

## Description 

This PR adds three new components: 
- `ItemsList`: A component responsible for retrieving and displaying the database collection documents on the screen.
- `AddItemsForm`: A component responsible for making changes to the database.
- `Home`: A component works as a main page and a container for both `ItemsList` and `AddItemsForm` components.

### Display items from the database 

Inside that `ItemsList` component, we've used `FirestoreCollection` which is a component that allows you to interact with a Firestore collection. Using this component, you can access the collection at a given path and provide sort options, perform queries, and paginate data.

This component will set up a listener and update whenever our collection is updated in Firestore.

More documentation on `FirestoreCollection` [here](https://www.npmjs.com/package/react-firestore#firestorecollection). 

### Write items to the database

To add items to the database we declared a simple function: `adddToDatabase()` function. Inside that function, we interacted with the `items` collection in the database through `firebase.firestore()` and then passed all the properties we needed to store as a new document in the `items` collection.

### How it looks so far (super ugly)
a form and a list of a few items retrieved from the database

## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin bo-hn-connect-firebase` and check that branch out with `git checkout bo-hn-connect-firebase` 
* Then `yarn` to install the newly added dependencies locally and `yarn start` to launch the app.
* Fill out the form with a new item. Make sure it adds it to the list and make sure that the item appears in the `items` collection of the database [here](https://console.firebase.google.com/u/0/project/tcl-15-smart-shopping-list/firestore/data~2F).

### CODE REVIEW

**Comment by deboragaleano** [comment]
It looks great to me! Everything worked fine on the browser and the changes were reflected on the DB as well. 🎉

File: src/components/AddItemsForm.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react';
+import { items } from '../lib/firebase';
+import '../css/components/AddItemsForm.css';
+
+const AddItemsForm = () => {
+  let [inputValue, setInputValue] = useState('');
+
+  const addToDatabase = (e) => {
+    e.preventDefault();
+    items()
```
Comment (yahayakenny) [inline src/components/AddItemsForm.js:10]: Here, could you wrap the code from lines 10-18 in a 'try' block before the 'catch' block on line 19. I dont know if that's possible?

Comment (saraheisa) [reply src/components/AddItemsForm.js:10]: Hi @yahayakenny thanks for your review ✨
and It's a promise, not a try/catch block

Comment (yahayakenny) [reply src/components/AddItemsForm.js:10]: Hi Sarah, yeah, I just went ahead and read a bit about it and also discovered that. You guys have some really great work here. Everything works perfectly. Good job  👍

File: src/components/AddItemsForm.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      .add({
+        name: inputValue,
+        added_on: Date.now(),
+      })
+      .then((res) => {
+        setInputValue('');
```
Comment (llunaplanet) [inline src/components/AddItemsForm.js:16]: Yeah, this is a nice detail to clean the input filed after a successful add to the database :heart:

**Review by llunaplanet — APPROVED** [review]
Ok, all checked, looks good to me! Let's merge this!

File: src/components/ItemsList.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ function ItemsList() {
   return (
     <FirestoreCollection
       path="items"
-      sort="added_on:desc"
+      // sort="added_on:desc"  => disable sorting for now
```
Comment (llunaplanet) [inline src/components/ItemsList.js]: I love the fact that you have been playing with the collection sorting, that is great!, the only thing is that it's adding a little bit of noise for the reviewer. I mean that we should try to stick to the AC, for example, with git, once we are happy with our PR, we can create a new branch from it to explore and experiment.

**Comment by llunaplanet** [comment]
Loving the use of **refactor:** prefix in the latest commits, that is a very good practice and helps a lot. There is an initiative to try to standardize that a bit, it's called https://www.conventionalcommits.org/en/v1.0.0/

**Pull request merged**